src/components/Reports/FourteenMonthReports/useFourteenMonthReport.ts

```diff
@@ -0,0 +1,127 @@
+import { useMemo } from 'react';
+import {
+  GetFourteenMonthReportQuery,
+  useGetFourteenMonthReportQuery,
+} from 'src/components/Reports/FourteenMonthReports/GetFourteenMonthReport.generated';
+import { convertStatus } from 'src/utils/functions/convertContactStatus';
+import { FourteenMonthReportCurrencyType } from './FourteenMonthReport';
+
+type CurrencyGroups = NonNullable<
+  GetFourteenMonthReportQuery['reportsSalaryCurrencyDonations']
+>['currencyGroups'];
+
+export const useFourteenMonthReport = (
+  accountListId: string,
+  currencyType: FourteenMonthReportCurrencyType,
+  designationAccounts?: string[],
+) => {
+  const {
+    data: gqlData,
+    loading,
+    error,
+  } = useGetFourteenMonthReportQuery({
+    variables: {
+      accountListId,
+      // Backend sends one extra month, e.g. 13m returns 14 months
+      range: '13m',
+      designationAccountId:
+        designationAccounts && designationAccounts.length > 0
+          ? designationAccounts
+          : null,
+    },
+  });
+
+  const data =
+    currencyType === FourteenMonthReportCurrencyType.Salary
+      ? gqlData?.reportsSalaryCurrencyDonations
+      : gqlData?.reportsDonorCurrencyDonations;
+
+  const fourteenMonthReport = useMemo(() => {
+    if (!data) {
+      return null;
+    }
+
+    const isSalaryType =
+      currencyType === FourteenMonthReportCurrencyType.Salary;
+    const currencyGroups: CurrencyGroups = data.currencyGroups;
+
+    return {
+      currencyType,
+      salaryCurrency: data.defaultCurrency ?? 'USD',
+      currencyGroups: Object.entries(currencyGroups ?? {}).map(
```

Comment:
Since the `currencyGroups` type isn't nullable (it is `JSON!`), I don't think we need to handle missing values.

```suggestion
      currencyGroups: Object.entries(currencyGroups).map(
```